static/include/services/apis/generated/data_manager.md

```diff
@@ -0,0 +1,119 @@
+### Sync
+
+{{% alert title="Important" color="tip" %}}
+
+This method is not yet available in the Viam Python SDK.
+
+{{% /alert %}}
+
+Sync data stored on the machine to the cloud.
+
+{{< tabs >}}
+{{% tab name="Go" %}}
+
+**Parameters:**
+
+- `ctx` [(Context)](https://pkg.go.dev/context#Context): A Context carries a deadline, a cancellation signal, and other values across API boundaries.
+- `extra` [(map[string]interface{})](https://go.dev/blog/maps): Extra options to pass to the underlying RPC call.
+
+**Returns:**
+
+- [(error)](https://pkg.go.dev/builtin#error): An error, if one occurred.
+
+**Example:**
+
+```go {class="line-numbers linkable-line-numbers"}
+// Sync data stored on the machine to the cloud.
+err := data.Sync(context.Background(), nil)
+```
+
+For more information, see the [Go SDK Docs](https://pkg.go.dev/go.viam.com/rdk/services/datamanager#Service).
+
+{{% /tab %}}
+{{< /tabs >}}
+
+### Reconfigure
+
+Reconfigure this resource.
+Reconfigure must reconfigure the resource atomically and in place.
+
+{{< tabs >}}
+{{% tab name="Go" %}}
+
+**Parameters:**
+
+- `ctx` [(Context)](https://pkg.go.dev/context#Context): A Context carries a deadline, a cancellation signal, and other values across API boundaries.
+- `deps` [(Dependencies)](https://pkg.go.dev/go.viam.com/rdk/resource#Dependencies): The resource dependencies.
+- `conf` [(Config)](https://pkg.go.dev/go.viam.com/rdk/resource#Config): The resource configuration.
+
+**Returns:**
+
+- [(error)](https://pkg.go.dev/builtin#error): An error, if one occurred.
+
+For more information, see the [Go SDK Docs](https://pkg.go.dev/go.viam.com/rdk/resource#Resource).
+
+{{% /tab %}}
+{{< /tabs >}}
+
+### DoCommand
+
+Execute model-specific commands that are not otherwise defined by the service API.
+For built-in service models, any model-specific commands available are covered with each model's documentation.
+If you are implementing your own data manager service and add features that have no built-in API method, you can access them with `DoCommand`.
+
+{{< tabs >}}
+{{% tab name="Go" %}}
+
+**Parameters:**
+
+- `ctx` [(Context)](https://pkg.go.dev/context#Context): A Context carries a deadline, a cancellation signal, and other values across API boundaries.
+- `cmd` [(map[string]interface{})](https://go.dev/blog/maps): The command to execute.
+
+**Returns:**
+
+- [(map[string]interface{})](https://pkg.go.dev/builtin#string): The command response.
+- [(error)](https://pkg.go.dev/builtin#error): An error, if one occurred.
+
+**Example:**
+
+```go {class="line-numbers linkable-line-numbers"}
+// This example shows using DoCommand with an arm component.
```

Comment:
Is there a realistic way in which we imagine someone could use DoCommand with the data service? It seems that, while we use arm component for all the component examples which is already a bit of a stretch, using an arm component as a stand-in for the data service is perhaps such a dissimilar example that it's unhelpful? Maybe something with generic service for all services? Also, are there any services that do not support Do?

Reply:
No, all resources that inherit from resource.* (in this case [resource.Resource](https://pkg.go.dev/go.viam.com/rdk@v0.30.0/resource#Resource)) must share the same code sample for those inherited methods.

No there are not.